app/furniture/marketplace/management_component.rb

```diff
@@ -7,5 +7,17 @@ def initialize(marketplace:, **kwargs)
 
       self.marketplace = marketplace
     end
+
+    def button(location, icon:)
+      label, href = if location.is_a?(Symbol)
+        [t("marketplace.marketplace.#{location}.link_to"),
+          marketplace.location(location)]
+      else
+        [t("marketplace.#{location[:child]}.index.link_to"),
+          marketplace.location(**location)]
+      end
+
+      ButtonComponent.new(label: label, icon: icon, href: href, turbo_stream: false, method: :get, scheme: :secondary)
+    end
```

Comment:
❓ Could this be a separate component called `MarketplaceButton`?

I'm just becoming familiar with the ViewComponent framework and patterns, so you might have to school me here. But I don't understand the advantage of using this "container" class. Could you achieve the same thing with getting the marketplace instance from the controller and passing it to a `MarketplaceButton` like so? 

```erb
<%= render MarketPlaceButton(marketplace: marketplace, :edit, icon: :gear) if policy(marketplace).edit? %>
```

In JavaScript frontends/frameworks I've worked in we commonly build up domain-specific components from base components this way.

Reply:
I could see a very tiny `Marketplace::ManagementComponent::ButtonComponent` being useful for encapsulating this, but I don't see how that is substantively better than the helper method at this time.

Basically in my encapsulation-journey I tend to reach for things in the following order:

- a `variable` or `loop`
-  a `method` 
- a `Class`

At this point, the complexity doesn't feel significant enough to justify a class;; especially since the whole ruby-side of the component is < 100 lines at this point.